static-analysis/teammates-pmd.xml

```diff
@@ -27,6 +27,8 @@
         <exclude name="SimpleDateFormatNeedsLocale"/>
         <exclude name="UseLocaleWithCaseConversions"/>
         <exclude name="UseVarargs"/>
+        <exclude name="DataClass"/>
```

Comment:
If you have done enough research, you would know that PMD's rules grouping is entirely rewritten. That means you need to find out what rules were actually enforced before, see under which group they belong to now, and enforce/suppress accordingly.

Reply:
I did know about that. My aim was first to get the build to pass, then work my way up by writing the rules according to the new file structures. (I was hesitant to make those changes before getting a review from a senior developer first :sweat_smile: ). And more over I thought of dealing with the major changes which were causing the build failures and then making those other changes (As even though the grouping has changed entirely , every rule in those old ruleset files refer to the new rulesets only, even if this is for the time being)

Reply:
Anyway I will make those changes accordingly.